pyk/src/pyk/konvert/_module_to_kore.py

```diff
@@ -463,16 +469,14 @@ def functional_axiom(production: KProduction) -> Axiom:
         )
 
     res: list[Axiom] = []
-    for sentence in module.sentences:
-        if not isinstance(sentence, KProduction):
+    for prod in module.productions:
+        if not prod.klabel:
             continue
-        if not sentence.klabel:
+        if prod.klabel.name in BUILTIN_LABELS:
             continue
-        if sentence.klabel.name in BUILTIN_LABELS:
+        if not Atts.FUNCTIONAL in prod.att:
             continue
-        if not Atts.FUNCTIONAL in sentence.att:
-            continue
-        res.append(functional_axiom(sentence))
+        res.append(functional_axiom(prod))
     return res
```

Comment:
There is a clear pattern of repeated code here: 
* for each production `prod` in the module
   - if not `prod.klabel`: skip
   - if `prod.klabel.name in BUILTIN_LABELS`: skip
   - if `TARGET in prod.att`: call `TARGET_axiom` (some instances have different criteria, sometimes several ones)

Since each productions should be part of exactly one group with a certain criterion, would it make sense to iterate over all productions just once, to classify them into `TARGET` groups and  and dispatch to each `TARGET_axiom` in just one place?
(That place would be https://github.com/runtimeverification/k/pull/4704/files#diff-0e99d64ff2ae75d6aa4d5eb3c1822e330e1f2d1f8b69fb19bbb399e5780e2591R107-R119 )